mnist scenario

avssridhar opened the pull request:
Developed the implementation for mnist classification.

MNIST set is a large collection of handwritten digits. It is a very popular dataset in the field of image processing. It is often used for benchmarking machine learning algorithms. MNIST is short for Modified National Institute of Standards and Technology database. We here consider a hypothetical scenario where in we have three TDP's and the MNIST data is divided into three parts each coming from one TDP and TDC who wishes the train a model using datasets from these TDPs. The repository contains sample datasets and a model. The model and datasets are for illustrative purposes only; none of these organizations have been involved in contributing to the code or datasets.


**Tests Performed**

./ci/build.sh
cd scenarios/mnist
./ci/build.sh

cd scenarios/mnist/deployment/docker
./preprocess.sh

./save-model.sh

./train.sh

avssridhar committed:
mnist scenario

avssridhar committed:
more changes

avssridhar committed:
more changes

avssridhar committed:
more changes

avssridhar committed:
more changes

avssridhar committed:
more changes

avssridhar committed:
Update README.md

avssridhar committed:
Merge branch 'iSPIRT:main' into main

avssridhar committed:
Update Dockerfile.mnist_1

avssridhar committed:
Update push-containers.sh

avssridhar committed:
Update push-containers.sh

avssridhar committed:
Update query_config.json

avssridhar committed:
Update contract.json

avssridhar committed:
Update deploy.sh

avssridhar committed:
Update arm-template.json

avssridhar committed:
Update 5-upload-encrypted-data.sh

avssridhar committed:
Update 3-import-keys.sh

avssridhar committed:
Update 1-create-storage-containers.sh

avssridhar committed:
Delete .DS_Store

avssridhar committed:
Delete scenarios/.DS_Store

avssridhar committed:
Delete scenarios/mnist/.DS_Store

kapilvgit commented on scenarios/mnist/README.md:
````diff
@@ -0,0 +1,219 @@
+The script joins the datasets using a configuration defined in [query_config.json](./config/query_config.json) and trains the model using a configuration defined in [model_config.json](./config/model_config.json). If all goes well, you should see output similar to the following output, and the trained model will be saved under the folder `/tmp/output`. 
+
+```
+docker-train-1  | {'input_dataset_path': '/tmp/sandbox_icmr_cowin_index_without_key_identifiers.csv', 'saved_model_path': '/mnt/remote/model/model.onnx', 'saved_model_optimizer': '/mnt/remote/model/dpsgd_model_opimizer.pth', 'saved_weights_path': '', 'batch_size': 2, 'total_epochs': 5, 'max_grad_norm': 0.1, 'epsilon_threshold': 1.0, 'delta': 0.01, 'sample_size': 60000, 'target_variable': 'icmr_a_ ...
````
Please fix this comment, and replace with a sample log from mnist training.

kapilvgit commented on scenarios/mnist/config/query_config.json:
```diff
@@ -0,0 +1,45 @@
+   ],
+   "joined_dataset": {
+      "joined_dataset": "sandbox_mnist_without_key_identifiers.csv",
+      "joining_query": "",
```
Why are these empty? It's not good practice to use empty to default to some behavior. Let's try and make the operation explicit.

avssridhar replied:
Agreed. Since the mnist scenario does not technically involve a join this was left empty. Will replace it with a mock query.

kapilvgit replied:
It shouldnt be a mock query....can we have a query that does the aggregation?

avssridhar replied:
yes. Replacing it with a query which does union of all the three mnist partitions.

kapilvgit commented on scenarios/mnist/contract/contract.json:
```diff
@@ -0,0 +1,93 @@
+      "key" : {
+        "type": "azure",
+        "properties": {
+          "kid": "MNIST_1_FilesystemEncryptionKey",
```
kid is not consistent as in the previous PR

avssridhar replied:
making the kid consistent "MNIST_1_FilesystemEncryptionKey"

kapilvgit commented on scenarios/mnist/data/3-import-keys.sh:
```diff
@@ -0,0 +1,56 @@
+fi
+
+DATADIR=`pwd`
+import_key "MNIST_1FilesystemEncryptionKey" $DATADIR/mnist_1key.bin
```
This keyid must match the key id you are using in the other file.

kapilvgit commented on scenarios/mnist/data/output/ccr_depa_trg_model_logger.json:
Remove this file

kapilvgit commented on scenarios/mnist/deployment/docker/docker-compose-train.yml:
```diff
@@ -0,0 +1,12 @@
+  train:
+    image: ${CONTAINER_REGISTRY}/depa-training:latest
+    volumes:
+      - $ICMR_INPUT_PATH:/mnt/remote/icmr
```
These are not correct. Please make sure we have no references to icmr, cowin and index under the mnist folder.

kapilvgit commented on scenarios/mnist/deployment/docker/docker-compose-preprocess.yml:
```diff
@@ -0,0 +1,19 @@
+services:
+  icmr:
```
Wrong names

kapilvgit commented on scenarios/mnist/deployment/docker/train.sh:
```diff
@@ -0,0 +1,11 @@
+export DATA_DIR=$PWD/../../data
+export ICMR_INPUT_PATH=$DATA_DIR/icmr/preprocessed
```
Wrong names

kapilvgit commented on scenarios/mnist/policy/policy.rego:
Drop this file. We dont need it. I will drop the file in the covd scenario too.

avssridhar committed:
Update train.sh

changed_names

avssridhar committed:
Update README.md

added minist logs

avssridhar committed:
Delete scenarios/mnist/data/output/ccr_depa_trg_model_logger.json

deleted file

avssridhar committed:
Update docker-compose-train.yml

changed names

avssridhar committed:
Update docker-compose-preprocess.yml

changed names

avssridhar committed:
Delete scenarios/mnist/policy/policy.rego

deleted file

avssridhar committed:
Update contract.json

making the kid's consistent

avssridhar committed:
Update 3-import-keys.sh

making the KID consistent

kapilvgit commented on scenarios/mnist/data/.DS_Store:
Delete this file

kapilvgit commented on scenarios/mnist/data/4-encrypt-data.sh:
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
This also needs fixing

kapilvgit commented on scenarios/mnist/data/mnist_1/.DS_Store:
Delete this file

kapilvgit commented on scenarios/mnist/data/mnist_2/.DS_Store:
Delete this file

avssridhar committed:
Update query_config.json

adding the joining query field which does the union of the three mnist partitions.

avssridhar committed:
Update 4-encrypt-data.sh

changed names

kapilvgit committed:
fixes

avssridhar committed:
Merge pull request #2 from kapilvgit/main

fixes

avssridhar committed:
Update ccr_dpsgd_model_saving_template_v2.py

changed the model
